apps/chat/src/utils/app/conversation.ts

```diff
@@ -209,3 +216,30 @@ export const isChosenConversationValidForCompare = (
 
   return true;
 };
+
+export const getOpenAIEntityFullName = (model: OpenAIEntity) =>
+  skipEmptyAndJoin([model.name, model.version], '-') || model.id;
+
+interface ModelGroup {
+  groupName: string;
+  entities: OpenAIEntity[];
+}
+
+export const groupModelsAndSaveOrder = (
+  models: OpenAIEntity[],
+): ModelGroup[] => {
+  const uniqModels = uniq(models);
```

Comment:
Looks like this is bad check for unique objects - they just check for object reference
Or it's intended?

Reply:
it's correct check. Actually we have array like `[...recentModels, ...allModels]` and to avoid duplications I added `uniq`